tower-http/src/compression/layer.rs

```diff
@@ -31,48 +33,145 @@ impl CompressionLayer {
     /// Sets whether to enable the gzip encoding.
     #[cfg(feature = "compression-gzip")]
     #[cfg_attr(docsrs, doc(cfg(feature = "compression-gzip")))]
-    pub fn gzip(self, enable: bool) -> Self {
+    pub fn gzip(mut self, enable: bool) -> Self {
```

Comment:
Why are all these `mut`s necessary?

Reply:
This is the bug exactly. The next line `self.accept.set_gzip(enable)` wouldn't do anything. Because `set_gzip` was an adaptor instead of a setter, it returns a new AcceptEncoing here and is immediately dropped. Thus the configuration fails to register.

See below code change in `AcceptEncoding`'s struct. `AcceptEncoing::set_gzip` is now a setter instead of an adaptor, see below. the mut here is necessary because the next line mutates self.

Reply:
Makes sense.